code/modules/power/power.dm

```diff
@@ -262,6 +251,10 @@
 		else
 			continue
 
+	//Actually apply all the shit
+	for(var/obj/structure/cable/C in worklist)
+		PN.add_cable(C)
```

Comment:
doing all the application at the end like this may also be the source for lagspikes. It technically is only needed for very specific situations so I may make it a parameter as to where this is fired.

Reply:
ok yeah that was actually the problem.